lib/flop_phoenix.ex

```diff
@@ -467,6 +466,101 @@ defmodule Flop.Phoenix do
     """
   end
 
+  @doc """
+  Generates a filter input for a field.
+
+  ## Assigns
+
+  - `form` - `t:Phoenix.HTML.FormData`
+  - `label` (optional) - label to display, defaults to field name
+  - `using` - helper to render primary input field
+  - `op_selectable` (optional) - defines whether operation can be selected from
+    list, defaults to `false`
+  - `op_selectable_from` (optional) - list of `t:Flop.Filter.op` operations to
+    choose from
+  """
+  @spec filter_input(map) :: Phoenix.LiveView.Rendered.t()
+  def filter_input(assigns) do
+    assigns = Filter.init_input_assigns(assigns)
+
+    ~H"""
+    <div>
+      <%= hidden_input @form, :field %>
```

Comment:
re: `filters[]` -> I think this is why I'm still confused about the `filter_input` vs `filter_inputs_for` approach, in particular how these two public APIs would combine in practise.

If you have a form like this:

```
<input name="filters[][field]" value="foo">
<input name="filters[][op]" value="==">
<input name="filters[][value]" value="bar">
<input name="filters[][field]" value="baz">
<input name="filters[][op]" value="==">
<input name="filters[][value]" value="quuz">
```

the params Phoenix makes of this are 

```
%{"filters" => [%{"field" => "foo"}, %{"op" => "=="}, %{"value" => "bar"}, %{"field" => "baz"}, %{"op" => "=="}, %{"value" => "quuz"}]}
```

which is not what we want. We want a form like this:

```
<input name="filters[0][field]" value="foo">
<input name="filters[0][op]" value="==">
<input name="filters[0][value]" value="bar">
<input name="filters[1][field]" value="baz">
<input name="filters[1][op]" value="==">
<input name="filters[1][value]" value="quuz">
```

... which of course we can also achieve for `filter_input/1` by giving it a manual `index` option, so

```
<.filter_input form={f} field="foo" index="0" meta={meta} />
```

renders as the first the lines above (by looking up the `"bar"` value in the `meta`). But I doubt this is what you had in mind? We can totally follow through with this approach, but TBH then the outer form doesn't need to use `inputs_for` anymore as we entirely bypass its internal enumeration+naming logic, as well as the data fetching. 

In other words, the more I think about it, the less is the `filter_input` API you have in mind a part of `filter_inputs_for`, it basically doesn't have use for it when it uses `Phoenix.HTML.Form.inputs_for`.

---

May I suggest the following: We continue with `filter_inputs_for` and *don't* make `filter_input` a public API for now, to focus on one of the envisioned ways to create these forms. And we revisit a low-level public `filter_input` once we have all the associated (shared) boilerplate in place? I think this would simplify the first iteration of this quite a bit.